### ISSUE TITLE
Test proposal

### ISSUE THREAD

**Issue by maelle** [ropensci-review-tools/pkgcheck#68]
* Adding a general test like test-pkgcheck.R but on a package that does everything wrong. Snapshot for Markdown but also for the printed stuff.
* Testing each helper on its own: if it operates on a path, create skeleton and tweak it; all checks operating on pkgstats output could be tested in the same file. these tests would be much more minimal.

Thoughts?

**Comment by mpadge** [comment]
Yes, good and important ideas. I generally design tests to only call exported functions, and try to avoid direct testing of internal functions as much as possible. But this package, and all of the internal test functions in particular, is a good argument for direct testing of those. And designing and testing an epic-fail package is obviously also important.

**Comment by maelle** [comment]
As a check contributor, I'd find it less daunting if I could start with a really small test and then I'd include my check to the machinery (and it'd turn up in the less unit unit test).

**Comment by mpadge** [comment]
Addressing #40 and associated changes has gone some way towards this. All `pkgchk_` functions now take the same much more standardised input. That input has not yet been documented, but that's very high on priority list, and once that's done, this issue can then be addressed much more cleanly. It's all getting there ...

**Comment by mpadge** [comment]
Oh yeah, thanks. This has now been addressed:

>     * Testing each helper on its own: if it operates on a path, create skeleton and tweak it; all checks operating on pkgstats output could be tested in the same file. these tests would be much more minimal.

Each check is now expected to have its own separate tests, and all can use skeleton-generation helper functions, like this:
https://github.com/ropensci-review-tools/pkgcheck/blob/85124b0ffb3e9283bf530e100c241f17b9928973/tests/testthat/test-check-scrap.R#L7-L9

And the general procedure is [described in a vignette](https://docs.ropensci.org/pkgcheck/articles/extending-checks.html#a-testing-new-checks). So I think we can close.
